core/src/banking_stage/transaction_scheduler/greedy_scheduler.rs

```diff
@@ -0,0 +1,738 @@
+use {
+    super::{
+        in_flight_tracker::InFlightTracker,
+        prio_graph_scheduler::{
+            Batches, PrioGraphScheduler, TransactionSchedulingError, TransactionSchedulingInfo,
+        },
+        scheduler::{Scheduler, SchedulingSummary},
+        scheduler_error::SchedulerError,
+        thread_aware_account_locks::{ThreadAwareAccountLocks, ThreadId, ThreadSet},
+        transaction_priority_id::TransactionPriorityId,
+        transaction_state::{SanitizedTransactionTTL, TransactionState},
+        transaction_state_container::StateContainer,
+    },
+    crate::banking_stage::{
+        consumer::TARGET_NUM_TRANSACTIONS_PER_BATCH,
+        read_write_account_set::ReadWriteAccountSet,
+        scheduler_messages::{ConsumeWork, FinishedConsumeWork, TransactionBatchId},
+    },
+    crossbeam_channel::{Receiver, Sender, TryRecvError},
+    itertools::izip,
+    solana_cost_model::block_cost_limits::MAX_BLOCK_UNITS,
+    solana_runtime_transaction::transaction_with_meta::TransactionWithMeta,
+    solana_sdk::saturating_add_assign,
+};
+
+pub(crate) struct GreedySchedulerConfig {
+    pub max_scheduled_cus: u64,
+    pub max_scanned_transactions_per_scheduling_pass: usize,
+    pub target_transactions_per_batch: usize,
+    pub stop_on_unschedulable: bool,
+}
+
+impl Default for GreedySchedulerConfig {
+    fn default() -> Self {
+        Self {
+            max_scheduled_cus: MAX_BLOCK_UNITS,
+            max_scanned_transactions_per_scheduling_pass: 1000,
+            target_transactions_per_batch: TARGET_NUM_TRANSACTIONS_PER_BATCH,
+            stop_on_unschedulable: false,
```

Comment:
The greedy scheduler that was tested on mainnet so far had this set to **true**. This is different than the default here. Reason for the change - we want to schedule things that are deeper in our queue so that local fee markets work.

@alessandrod we probably want have a higher scan limit here since we aren't building a graph and can scan much faster. Let's coordinate to find a good default.